docs/content/doc/advanced/hacking-on-gitea.en-us.md

```diff
@@ -15,41 +15,258 @@ menu:
+from source</a>.
+
+The simplest recommended way to build from source is:
+
+```bash
+TAGS="bindata sqlite sqlite_unlock_notify" make generate build
 ```
-git remote rename origin upstream
-git remote add origin git@github.com:<USERNAME>/gitea.git
-git fetch --all --prune
+
+However, there are a number of additional make tasks you should be aware of.
+These are documented below but you can look at our
+[`Makefile`](https://github.com/go-gitea/gitea/blob/master/Makefile) for more,
+and look at our
+[`.drone.yml`](https://github.com/go-gitea/gitea/blob/master/.drone.yml) to see
+how our continuous integration works.
+
+### Formatting, linting, vetting and spell-check
+
+Our continous integration will reject PRs that are not properly formatted, fail
+linting, vet or spell-check.
+
+You should format your code with `go fmt` using:
+
+```bash
+make fmt
+```
+
+and can test whether your changes would match the results with:
+
+```bash
+make fmt-check # which runs make fmt internally
+```
+
+**Note**: The results of `go fmt` are dependent on the version of `go` present.
+You should run the same version of go that is on the continuous integration
+server as mentioned above. `make fmt-check` will only check if your `go` would
+format differently - this may be different from the CI server version.
+
+You should lint, vet and spell-check with:
+
+```bash
+make vet lint misspell-check
+```
+
+### Updating the stylesheets
+
+At present we use [less](http://lesscss.org/) to generate our stylesheets. Do
+**not** edit the files in `public/css/` directly as they are generated from
+`lessc` from the files in `public/less/`.
+
+If you wish to work on the stylesheets you will need to install `lessc` the
+less compiler. The recommended way to do this is using `npm install`:
+
+```bash
+cd "$GOPATH"/src/code.gitea.io/gitea
+npm install
+```
+
+You can then edit the less stylesheets and regenerate the stylesheets using:
+
+```bash
+make generate-stylesheets
 ```
 
-This should provide a working development environment for Gitea. Take a look at
-the `Makefile` to get an overview about the available tasks. The most common tasks
-should be `make test` which will start our test environment and `make build` which
-will build a `gitea` binary into the working directory. Writing test cases is not
-mandatory to contribute, but it is highly encouraged and helps developers sleep
-at night.
+You should commit both the changes to the css and the less files when making
+PRs.
+
+### Updating the API
+
+When creating new API routes or modifying existing API routes you **MUST**
+update and/or create [Swagger](https://swagger.io/docs/specification/2-0/what-is-swagger/)
+documentation for these. You should be careful not to break the API for
+downstream users which depend on a stable API. In general this means additions
+are acceptable, but deletions or fundamental changes of API will be rejected.
+
+Once you have created or changed an API endpoint, please regenerate the Swagger
```

Comment:
We should also add a link to the library we use to genereate the docs from comments and about the comment syntax.

Reply:
Do you mean more swagger documentation or something else?

Reply:
Maybe link to https://goswagger.io/use/spec.html#annotation-syntax

Reply:
Or a PR that add an endpoint: https://github.com/go-gitea/gitea/pull/5843/files